k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/selectedstate/command/UidSearchCommand.java

```diff
@@ -0,0 +1,191 @@
+package com.fsck.k9.mail.store.imap.selectedstate.command;
+
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+import com.fsck.k9.mail.Flag;
+import com.fsck.k9.mail.MessageRetrievalListener;
+import com.fsck.k9.mail.MessagingException;
+import com.fsck.k9.mail.store.imap.Commands;
+import com.fsck.k9.mail.store.imap.ImapConnection;
+import com.fsck.k9.mail.store.imap.ImapFolder;
+import com.fsck.k9.mail.store.imap.ImapMessage;
+import com.fsck.k9.mail.store.imap.ImapResponse;
+import com.fsck.k9.mail.store.imap.ImapUtility;
+import com.fsck.k9.mail.store.imap.selectedstate.response.UidSearchResponse;
+
+
+public class UidSearchCommand extends SelectedStateCommand {
```

Comment:
note that those don't have to be fancy: just call the setters and see that it returns the expected string. that already covers a lot of ground